fault/logging.py

```diff
@@ -0,0 +1,50 @@
+"""
+Note(rsetaluri): This file is taken almost verbatim from magma/magma/logging.py.
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import logging
+import traceback
+import inspect
+import sys
+
+
+log = logging.getLogger("fault")
+
+
+def get_original_wire_call_stack_frame():
+    for frame in inspect.stack():
+        if sys.version_info < (3, 5):
+            function = inspect.getframeinfo(frame[0]).function
+        else:
+            function = frame.function
+        if function not in ["wire", "connect",
+                            "get_original_wire_call_stack_frame",
+                            "error", "warn"]:
+            break
+    if sys.version_info < (3, 5):
+        return frame[0]
+    else:
+        return frame.frame
```

Comment:
I don't think we need this extra logic for wire tracebacks. Can we remove this function and simplify `error` to basically just printout to stderr? we could just have it use `log.error` like the other functions.